### PR TITLE
fix: fix sequence bug

### PR DESCRIPTION
**Pull request by chuntaojun**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by AlexStocks — APPROVED** [review]

File: pkg/runtime/namespace/namespace.go [hunk 1]
```diff
@@ -38,11 +38,18 @@ import (
 	"github.com/arana-db/arana/pkg/util/log"
 )
 
-var _namespaces sync.Map
+var (
+	_namespaces sync.Map
+	_tmpns      sync.Map
```
Comment (dongzl) [inline pkg/runtime/namespace/namespace.go:43]: what's the meaning of `_tmpns`? suggest make a meaningful name.

Comment (chuntaojun) [reply pkg/runtime/namespace/namespace.go:43]: forget remove it, will fix latter

File: pkg/proto/sequence.go [hunk 2]
```diff
@@ -20,44 +20,81 @@ package proto
...
+func RegisterSequenceManager(l SequenceManager) {
+	_defaultSequenceManager = l
 }
 
-type SequenceConfig struct {
-	Name   string
-	Type   string
-	Option map[string]string
+func LoadSequenceManager() SequenceManager {
+	cur := _defaultSequenceManager
```
Comment (sanxun0325) [inline pkg/proto/sequence.go:39]: You can directly `return _defaultSequenceManager`

File: pkg/runtime/namespace/namespace.go (hunk 1 above)
Comment (jjeffcaii) [inline pkg/runtime/namespace/namespace.go:43]: Why?

File: pkg/proto/sequence.go [hunk 3]
```diff
@@ -20,44 +20,81 @@ package proto
...
+	// Sequence represents a global unique id generator.
+	Sequence interface {
+		// Acquire generates a next value in int64.
+		Acquire(ctx context.Context) (int64, error)
+		Reset() error
+		Update() error
+	}
+
+	// EnhanceSequence represents a global unique id generator.
+	EnhanceSequence interface {
```
Comment (jjeffcaii) [inline pkg/proto/sequence.go:71]: typo: Enhance is `vt`, should be Enhanced(`adj`).

File: pkg/sequence/sequence.go [hunk 4]
```diff
@@ -19,84 +19,153 @@ package sequence
...
+func init() {
+	proto.RegisterSequenceManager(newSequenceManager())
+}
 
-func NewSequenceManager() proto.SequenceManager {
+func newSequenceManager() proto.SequenceManager {
 	return &sequenceManager{
-		sequenceOptions:  make(map[string]proto.SequenceConfig),
-		sequenceRegistry: make(map[string]proto.EnhancedSequence),
+		lock:    sync.RWMutex{},
```
Comment (jjeffcaii) [inline pkg/sequence/sequence.go:37]: No need to initialize Mutex/RwMutex manually.

File: pkg/sequence/sequence.go [hunk 5]
```diff
@@ -19,84 +19,153 @@ package sequence
...
+	return t.schemas[schema]
+}
+
+func (t *tenantBucket) getOrCreate(schema string) *schemaBucket {
+	t.lock.Lock()
+	defer t.lock.Unlock()
+
+	if _, ok := t.schemas[schema]; !ok {
+		t.schemas[schema] = &schemaBucket{
+			lock:             sync.RWMutex{},
```
Comment (jjeffcaii) [inline pkg/sequence/sequence.go:66]: Same as above.

File: pkg/proto/sequence.go [hunk 6]
```diff
@@ -20,44 +20,81 @@ package proto
...
 )
 
-// SequenceSupplier Create the creator of Sequence
-type SequenceSupplier func() EnhancedSequence
-
 var (
-	// Record the list of Sequence plug -in through the registered self -registered
-	suppliersRegistry = make(map[string]SequenceSupplier)
+	Error_NotSequenceType  = errors.New("sequence type not found")
+	Error_NotFoundSequence = errors.New("sequence instance not found")
```
Comment (jjeffcaii) [inline pkg/proto/sequence.go:29]: Follow the Golang style.

**Review by jjeffcaii — CHANGES_REQUESTED** [review]
pls fix

**Review by jjeffcaii — APPROVED** [review]
LGTM

**Review by dongzl — APPROVED** [review]

File: pkg/proto/sequence.go [hunk 7]
```diff
@@ -20,10 +20,72 @@ package proto
...
+	"fmt"
 )
 
-// SequenceSupplier Create the creator of Sequence
-type SequenceSupplier func() EnhancedSequence
+var (
+	ErrorNotSequenceType  = errors.New("sequence type not found")
+	ErrorNotFoundSequence = errors.New("sequence instance not found")
+
+	_defaultSequenceManager SequenceManager
```
Comment (AlexStocks) [inline pkg/proto/sequence.go:31]: go 里面没有使用下划线的变量命名传统

Comment (chuntaojun) [reply pkg/proto/sequence.go:31]: 这里看整个arana项目差不多都是这样子的风格，所以就跟着一起了

**Review by sanxun0325 — APPROVED** [review]

File: pkg/proto/sequence.go (hunk 7 above)
Comment (jjeffcaii) [reply pkg/proto/sequence.go:31]: 这个ok的，参考uber的规范。https://github.com/uber-go/guide/blob/master/style.md#prefix-unexported-globals-with-_

**Pull request merged**